core/src/main/java/org/apache/calcite/sql/SqlTableFunction.java

```diff
@@ -30,4 +32,17 @@ public interface SqlTableFunction {
    * @return strategy to infer the row type of a call to this function
    */
   SqlReturnTypeInference getRowTypeInference();
+
+  /**
+   * Returns the table parameter characteristics for <code>ordinal</code>th
+   * parameter to this table function.
+   *
+   * <p>Returns <code>null</code> if the <code>ordinal</code>th argument is
+   * not table parameter or the <code>ordinal</code> is smaller than 0 or
+   * the <code>ordinal</code> is greater than or equals to the number of
+   * parameters.
+   */
+  default @Nullable TableCharacteristic tableCharacteristic(int ordinal) {
+    return null;
```

Comment:
Seems the functionality is only supported by window function ? Then moving this interface to `SqlWindowTableFunction` is more proper way.